example/HelloWorldFunction/src/main/java/helloworld/AppIdempotency.java

```diff
@@ -0,0 +1,85 @@
+package helloworld;
+
+import com.amazonaws.services.lambda.runtime.Context;
+import com.amazonaws.services.lambda.runtime.RequestHandler;
+import com.amazonaws.services.lambda.runtime.events.APIGatewayProxyRequestEvent;
+import com.amazonaws.services.lambda.runtime.events.APIGatewayProxyResponseEvent;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import software.amazon.lambda.powertools.idempotency.Idempotency;
+import software.amazon.lambda.powertools.idempotency.IdempotencyConfig;
+import software.amazon.lambda.powertools.idempotency.Idempotent;
+import software.amazon.lambda.powertools.idempotency.persistence.DynamoDBPersistenceStore;
+import software.amazon.lambda.powertools.utilities.JsonConfig;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class AppIdempotency implements RequestHandler<APIGatewayProxyRequestEvent, APIGatewayProxyResponseEvent> {
+    private final static Logger LOG = LogManager.getLogger();
+
+    public AppIdempotency() {
+        // we need to initialize idempotency configuration before the handleRequest method is called
+        Idempotency.config().withConfig(
+                        IdempotencyConfig.builder()
+                                .withEventKeyJMESPath("powertools_json(body).address")
```

Comment:
@jeromevdl could you explain what `"powertools_json(body).address"` means here.

Reply:
powertools_json will transform the body (which is a string) in a JSON object, and then we retrieve the address field inside this JSON.

Reply:
ah ok, so it's a reference to the `JsonFunction`